test/socket_test.dart

```diff
@@ -1,26 +1,176 @@
+      expect(socket.logger is Logger, true);
+      expect(socket.reconnectAfterMs is Function, true);
+    });
+
+    test('overrides some defaults with options', () async {
+      final socket = Socket('wss://example.com/socket',
+          timeout: const Duration(milliseconds: 40000),
+          longpollerTimeout: 50000,
+          heartbeatIntervalMs: 60000,
+          logger: (kind, msg, data) => print('[$kind] $msg $data'));
+      expect(socket.channels.length, 0);
+      expect(socket.sendBuffer.length, 0);
+      expect(socket.ref, 0);
+      expect(socket.endPoint, 'wss://example.com/socket/websocket');
+      expect(socket.stateChangeCallbacks, {
+        'open': [],
+        'close': [],
+        'error': [],
+        'message': [],
+      });
+      expect(socket.transport is IOWebSocketChannelClosure, true);
+      expect(socket.timeout, const Duration(milliseconds: 40000));
+      expect(socket.longpollerTimeout, 50000);
+      expect(socket.heartbeatIntervalMs, 60000);
+      expect(socket.logger is Logger, true);
+      expect(socket.reconnectAfterMs is Function, true);
+    });
+  });
+
+  group('endpointURL', () {
+    test('returns endpoint for given full url', () {
+      final socket = Socket('wss://example.org/chat');
+      expect(
+        socket.endPointURL(),
+        'wss://example.org/chat/websocket?vsn=1.0.0',
+      );
+    });
+
+    test('returns endpoint with parameters', () {
+      final socket = Socket('ws://example.org/chat', params: {'foo': 'bar'});
+      expect(socket.endPointURL(),
+          'ws://example.org/chat/websocket?foo=bar&vsn=1.0.0');
+    });
+
+    test('returns endpoint with apikey', () {
+      final socket =
+          Socket('ws://example.org/chat', params: {'apikey': '123456789'});
+      expect(socket.endPointURL(),
+          'ws://example.org/chat/websocket?apikey=123456789&vsn=1.0.0');
+    });
+  });
+
+  group('connect with Websocket', () {
+    Socket socket;
+
+    setUp(() {
+      socket = Socket('wss://localhost:0/');
+    });
+    test('establishes websocket connection with endpoint', () {
+      socket.connect();
+
+      final conn = socket.conn;
+
+      expect(conn, isA<IOWebSocketChannel>());
+      //! Not verifying connection url
+    });
+
+    test('sets callbacks for connection', () async {
+      int opens = 0;
+      socket.onOpen(() {
+        opens += 1;
+      });
+      int closes = 0;
+      socket.onClose((_) {
+        closes += 1;
+      });
+      dynamic lastErr;
+      socket.onError((e) {
+        lastErr = e;
+      });
+      dynamic lastMsg;
+      socket.onMessage((m) {
+        lastMsg = m;
+      });
+
+      socket.connect();
+      expect(opens, 1);
+
+      socket.conn.sink.addError('error');
+      expect(lastErr, 'error');
+
+      const data = {
+        'topic': 'topic',
+        'event': 'event',
+        'payload': 'payload',
+        'status': 'ok',
+      };
+      socket.conn.sink.add(data);
```

Comment:
This won't work. You need to encode the data, can't just pass it directly like that. I suggest you use `sendHeartbeat` method to test onMessage event.

```
socket.sendHeartbeat();
// need to wait for event to trigger
await Future.delayed(const Duration(seconds: 1), () {});
expect(lastMsg['event'], 'phx_heartbeat');
```